src/main.rs

```diff
@@ -103,20 +107,32 @@ impl FileId {
     }
 }
 
+#[derive(PartialEq)]
+enum WriteMode {
+    AskConfirm,
+    #[allow(dead_code)]
+    Force,
+}
+enum SenderEvents {
+    Listening(u16),
```

Comment:
Ah, this is a neat solution! It handles both the ordering and knowing the port number, nice!